inst/@sym/equationsToMatrix.m

```diff
@@ -0,0 +1,173 @@
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym equationsToMatrix (@var{eqns}, @var{vars})
+%% Convert set of linear equations to matrix form.
+%% Where @var{eqns} and @var{vars} can be a list or matrix of expressions.
+%%
+%% Example:
+%% @example
+%% @group
+%% syms x y z
+%% [A b] = equationsToMatrix (x + y == 1, x - y + 1, x, y)
+%%   @result{} A = (sym 2×2 matrix)
+%%       ⎡1  1 ⎤
+%%       ⎢     ⎥
+%%       ⎣1  -1⎦
+%%
+%%   b = (sym 2×1 matrix)
+%%
+%%       ⎡1 ⎤
+%%       ⎢  ⎥
+%%       ⎣-1⎦
+%%
+%% @end group
+%% @end example
+%%
+%% @example
+%% @group
+%% [A, B] = equationsToMatrix ([x + y - 2*z == 0, x + y + z == 1, 2*y - z + 5 == 0], [x, y])
+%%   @result{} A = (sym 3×2 matrix)
+%%
+%%       ⎡1  1⎤
+%%       ⎢    ⎥
+%%       ⎢1  1⎥
+%%       ⎢    ⎥
+%%       ⎣0  2⎦
+%%
+%%   B = (sym 3×1 matrix)
+%%
+%%       ⎡ 2⋅z  ⎤
+%%       ⎢      ⎥
+%%       ⎢-z + 1⎥
+%%       ⎢      ⎥
+%%       ⎣z - 5 ⎦
+%%
+%% @end group
+%% @end example
+%%
+%% @end defmethod
+
+
+function [A b] = equationsToMatrix(varargin)
+
+  s = symvar ([varargin{:}]);
+
+  cmd = {'vars = list()'
+         'if not isinstance(_ins[-2], MatrixBase):'
+         '    if isinstance(_ins[-2], Symbol):'
+         '        del _ins[-1]'
+         '        for i in reversed(range(len(_ins))):'
+         '            if isinstance(_ins[i], Symbol):'
+         '                vars = [_ins[i]] + vars'
+         '                del _ins[-1]'
+         '            else:'
+         '                break'
+         '    else:'
+         '        vars = _ins[-1]'
+         '        del _ins[-1]'
+         'else:'
+         '    if len(_ins) == 2:'
+         '        vars = _ins[-1]'
+         '        del _ins[-1]'
+         '    else:'
+         '        vars = _ins[-2]'
+         '        del _ins[-1]'
+         '        del _ins[-1]'
+         'vars = list(collections.OrderedDict.fromkeys(vars))' %% Never repeat elements
+         '_ins = list(flatten(_ins))' %% Unpack eqs
+         'if len(_ins) == 0 or len(vars) == 0:'
+         '    return True, Matrix([]), Matrix([])'
+         'A = zeros(len(_ins), len(vars)); b = zeros(len(_ins), 1)'
+         'for i in range(len(_ins)):' 
+         '    q = _ins[i]'
+         '    for j in range(len(vars)):'
+         '        p = Matrix(Poly.from_expr(_ins[i], vars[j]).all_coeffs())'
+         '        q = Poly.from_expr(q, vars[j]).all_coeffs()'
+         '        over = True if len(p) > 2 else False'
+         '        p = p[0] if len(p) == 2 else S(0)'
+         '        q = q[1] if len(q) == 2 else q[0]'
+         '        if not set(p.free_symbols).isdisjoint(set(vars)) or over:'
+         '            return False, 0, 0'
+         '        A[i, j] = p'
+         '    b[i] = -q'
+         'return True, A, b' };
+
+  [s A b] = python_cmd (cmd, sym (varargin){:}, s);
+
+
+  if ~s
+    error('Cannot convert to matrix form because the system does not seem to be linear.');
+  end
+
+end
+
+
+%%Matlab Tests
```

Comment:
We need to come up with our own tests.  Or we could take from a text book (which we cite).